### ISSUE TITLE
Add campaign badger

### ISSUE THREAD

**Issue by DeinFreund** [ZeroK-RTS/Zero-K-Infrastructure#1895]
I would like to add a badger that can be achieved by completing the campaign with all bonus objectives achieved on brutal. Of course there will still be difficulty adjustments, but I have the feeling that this might be enough of a challenge to warrant a badger. Akin to the level 200 or top3 badge, this shows that the player knows the in and outs of the game.

The campaign uses a different skill set from competitive play, so this should allow more people to gain access to precious badgers. Maybe it'll even cause other players to spend more time in the campaign (or install cheats) to get their badger.

Thoughts from @Aquanim ?

**Comment by Licho1** [comment]
I don't think it takes same time as level 200 or top 3 badge.. but sure there could be something..

**Comment by Anarchid** [comment]
Campaign is untrusted data. Anyone could just print that badge.

**Comment by DeinFreund** [comment]
> ... (or install cheats) to get their badger.

Cheating is always a possibility. As badgers are only a "passive" attribute that doesn't get you any special privileges or even on a ladder I don't think there would be much of a negative impact. Why cheat yourself a badger when you can use forced desync and speccheating to win every (recorded) multiplayer game. 

Maybe we could add a replay uploader to campaign, where you can optionally upload your replays as online battles. This would solve two problems at once: Campaign now allows you to gain some online xp similar to chicken and bot matches. Badgers are provable.

**Comment by GoogleFrog** [comment]
The missions I make seem to have particularly brutal brutal, I have only made two so far and only one is released. We're thinking of sweeping the galaxy with brutal updates when we're done.

It is quite easy to fake mission victories. The only communication channel that I know of from ingame to luamenu is via message sending in widgets and anyone could write a widget. This is only the second -easiest method to get progress. The campaign save file is a lua table sitting in a file.

I'm not against this proposal though. The badge will mostly be for the enjoyment of the player. We could have 4 badges, one for full completion on each difficulty level.

**Comment by Anarchid** [comment]
An alternative implementation of this item is a steam achievement.

**Comment by DeinFreund** [comment]
I prefer badges because they're designed for direct display in the game itself. We can add corresponding steam achievements for the badges easily.

**Comment by Aquanim** [comment]
I don't feel strongly about this one way or another.

**Comment by Licho1** [comment]
Closing this as it is mostly about deciding whether to do it .. if so, reopen, interface will be added to support this.

**Comment by GoogleFrog** [comment]
Why do you close tickets? This ticket is effectively about thinking.
